installers/msi-language/GAPixel/GAPixelCA.cs

```diff
@@ -5,59 +5,47 @@
 using System.Net;
 using Microsoft.Win32;
 
+
 namespace GAPixel
 {
     public class GetInfo
     {
 
-        private static Guid StringToGuid(string input)
-        {
-            using (System.Security.Cryptography.MD5 md5 = System.Security.Cryptography.MD5.Create())
-            {
-                byte[] inputBytes = System.Text.Encoding.ASCII.GetBytes(input);
-                byte[] hashBytes = md5.ComputeHash(inputBytes);
-
-                return new Guid(hashBytes);
-            }
-        }
-
-        public static string GetUniqueId(Session session=null)
+        public static string GetOrCreateNewCid(Session session)
         {
             try
             {
-                var oMClass = new ManagementClass("Win32_NetworkAdapterConfiguration");
-                var colMObj = oMClass.GetInstances();
-                foreach (var objMO in colMObj)
+                var baseKey = Registry.CurrentUser;
+                if (session.GetMode(InstallRunMode.Scheduled) && session.CustomActionData.ContainsKey("USERSID"))
```

Comment:
This will only get the registry key if this method is called from a deferred custom action. I _think_  this can be called from an immediate custom action via the `GAReport` CAs. If that's the case we can still try to get the registry key here we just have to use `Registry.CurrentUser`

Reply:
In the non-scheduled (immediate?) mode, `baseKey` is actually set to `Registry.CurrentUser` from which it then goes to the sub-key.  So, if I understand correctly, it is already implemented the way you describe it.

Reply:
Oh I see what I missed now. 👍